packages/entity/src/EntityConfiguration.ts

```diff
@@ -13,7 +13,7 @@ export default class EntityConfiguration<TFields> {
   readonly cacheableKeys: ReadonlySet<keyof TFields>;
   readonly cacheKeyVersion: number;
 
-  readonly getInboundEdges: () => IEntityClass<any, any, any, any, any, any>[];
+  readonly getInboundEdges: () => Promise<IEntityClass<any, any, any, any, any, any>[]>;
```

Comment:
API discussion topic: should we let developers write synchronous functions for getInboundEdges?

The advantage is that existing code that uses Entity doesn't need to be migrated. Also, direct synchronous calls produce cleaner stack traces and we can skip a tick of the async microtask queue.

The disadvantage is that there are now two ways to implement this method instead of just one. There's also a slight disadvantage in that the caller of this method might want to write: `if (typeof maybePromise.then === 'function') { await maybePromise }`, though it's also OK to write `await maybePromise` and non-promises will work fine.

```suggestion
  readonly getInboundEdges: () => IEntityClass<any, any, any, any, any, any>[] |
                                  Promise<IEntityClass<any, any, any, any, any, any>[]>;
```

Reply:
I think my preference is to keep types as strict as possible (and to have the types drive the runtime source of truth). In some repos I think something like this makes sense, but in this repo the types are as strict as possible.

As far as migrations go, large typescript migrations are pretty standard for this library as well.